services/api-server/src/simcore_service_api_server/api/routes/files.py

```diff
@@ -170,6 +182,117 @@ async def upload_files(files: list[UploadFile] = FileParam(...)):
     raise NotImplementedError
 
 
+@router.post(
+    "/content",
+    response_model=FileUploadSchema,
+)
+@cancel_on_disconnect
+@catch_n_raise(
+    (NodeportsException, ValidationError, ClientError),
+    lambda e: status.HTTP_500_INTERNAL_SERVER_ERROR,
+)
+async def get_upload_links(
+    request: Request,
+    client_file: ClientFile,
+    user_id: Annotated[PositiveInt, Depends(get_current_user_id)],
+):
+    """Get upload links for uploading a file to storage
+
+    Arguments:
+        request -- Request object
+        client_file -- ClientFile object
+        user_id -- User Id
+
+    Returns:
+        FileUploadSchema
+    """
+    assert request  # nosec
+    file_meta: File = await File.create_from_client_file(
+        client_file,
+        datetime.datetime.now(datetime.timezone.utc).isoformat(),
+    )
+    _, upload_links = await get_upload_links_from_s3(
+        user_id=user_id,
+        store_name=None,
+        store_id=SIMCORE_LOCATION,
+        s3_object=parse_obj_as(
+            StorageFileID, f"api/{file_meta.id}/{file_meta.filename}"
+        ),
+        client_session=None,
+        link_type=LinkType.PRESIGNED,
+        file_size=ByteSize(client_file.filesize),
+        is_directory=False,
+    )
+    return upload_links
+
+
+@router.patch(
+    "/content",
+    response_model=File,
+)
+@cancel_on_disconnect
+@catch_n_raise(
+    (ValueError, S3TransferError, ClientError, ValidationError),
+    lambda e: status.HTTP_500_INTERNAL_SERVER_ERROR,
+)
+async def complete_multipart_upload(
+    request: Request,
+    client_file: ClientFile,
+    uploaded_parts: FileUploadCompletionBody,
+    completion_link: FileUploadCompleteLinks,
+    user_id: Annotated[PositiveInt, Depends(get_current_user_id)],
+):
+    """Complete the multipart upload of a file
+
+    Arguments:
+        request: The Request
+        client_file: The ClientFile object
+        uploaded_parts: The uploaded parts
+        completion_link: The S3 completion link for this multipart upload
+        user_id: The user id
+
+    Returns:
+        The File object
+    """
+    assert request  # nosec
+    assert user_id  # nosec
+
+    e_tag: ETag = await complete_file_upload(
+        uploaded_parts=uploaded_parts.parts,
+        upload_completion_link=completion_link.state,
+    )
+
+    file_meta: File = await File.create_from_client_file(
+        client_file,
+        datetime.datetime.now(datetime.timezone.utc).isoformat(),
+        checksum=e_tag,
+    )
+    return file_meta
+
+
+@router.delete("/content")
```

Comment:
1. no need for ```include_in_schema=API_SERVER_DEV_FEATURES_ENABLED,```?
2. this should be more like DELETE /contents/{content_id}, otherwise it is not logical in terms of REST API and I guess it is questionable whether the upload *process* is a resource or not. It could also be POST /contents/{content_id}:abort

Reply:
Yes, I have also turned this into a `POST` method